### PR TITLE
In RealContentListingObject.__getattr__ check attribute existence without acquisition but return the attribute with acquisition in case it is a method that needs acquisition.

### PR DESCRIPTION
**Pull request by gbastien**
Hi @mauritsvanrees @baekholt 

See #47

I adapted code so `__getattr__` on a `contentlisting object` checks attribute existence on the object without acquisition but returns it with acquisition so when it is a method, calling it will have acquisition.

This change breaks `test_special_getattr_from_object`, but actually I think the test is wrong.  It does not defer to the brain, but actually calls `absolute_url` (OFS/Traversable.py(75)absolute_url()) and fails because object does not have acquisition.  You can `pdb` in `test_special_getattr_from_object` without my change to see that.
So I would adapt this test that will now return the `absolute_url`, but this changes the current behavior, so?

Thank you for review and feedback.

Gauthier

### CODE REVIEW

**Comment by gbastien** [comment]
@jenkins-plone-org please run jobs

File: plone/app/contentlisting/tests/test_integration_unit.py [hunk 1]
```diff
@@ -262,6 +266,14 @@ def test_item_get_none(self):
                 "result in an AttributeError."
             )
 
+    def test_item_getattr_acquisition(self):
+        self.portal.get("test-folder").setTitle("Test folder")
+        self.item._realobject.__class__.Title = patched_Title
```
Comment (davisagli) [inline plone/app/contentlisting/tests/test_integration_unit.py:271]: @gbastien This is being left in place after this test completes and breaking a lot of other tests. I'd suggest using https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch as a context manager to patch the Title method in a way that will be automatically removed whether or not there is an error in the test.

**Review by davisagli — COMMENTED** [review]
The change here seems sensible, but the test is failing to clean up after itself.

File: plone/app/contentlisting/tests/test_integration_unit.py (hunk 1 above)
Comment (gbastien) [reply plone/app/contentlisting/tests/test_integration_unit.py:271]: Hi @davisagli 

thank you for follow up.  Actually I could avoid the monkeypatch by using the absolute_url method that needs the acquisition to work as discovered by test test_special_getattr_from_object that is now succeeding in calling absolute_url.

I will adapt code and come back to you.

Thank you,

Gauthier

**Comment by gbastien** [comment]
Hi @davisagli @mauritsvanrees 

I take advantage of the PloneConf sprint to work on this issue.

I removed Title monkeypatch as the absolute_url method is there and uses acquisition.

Now the test_special_getattr_from_object is broken, the assumption of the test is wrong.

What should we do?

Thank you for follow up.

Gauthier

**Comment by gbastien** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by gbastien** [comment]
Hi @jensens @davisagli @mauritsvanrees 

as you can see, the `test_special_getattr_from_object` is broken "as expected".  It was correct before because acquisition was not there and absolute_url needs acquisition.

Thank you for review ;-)

Gauthier

**Comment by davisagli** [comment]
@gbastien Sorry, I see that you asked what to do with the test and no one answered. Please update the test to match the changed behavior.

**Comment by gbastien** [comment]
> @gbastien Sorry, I see that you asked what to do with the test and no one answered. Please update the test to match the changed behavior.

Right thank you I will do that!

**Comment by gforcada** [comment]
I can't really say how critical is this fix, but 6.0 is getting closer ™️ and if only a few iterations on the code are needed, maybe is a good time to try to finish it? 👍🏾

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**